app/scripts/models/mixins/resume-token.js

```diff
@@ -44,16 +54,47 @@ define(function (require, exports, module) {
     },
 
     /**
-     * Sets model properties from a ResumeToken.
+     * Sets model properties from a ResumeToken, unless said token
+     * is invalid according to `this.resumeTokenSchema`.
      *
      * @method populateFromResumeToken
      * @param {ResumeToken} resumeToken
      */
     populateFromResumeToken: function (resumeToken) {
       if (this.resumeTokenFields) {
-        this.set(resumeToken.pick(this.resumeTokenFields));
+        var pickedResumeToken = resumeToken.pick(this.resumeTokenFields);
+
+        if (isResumeTokenValid.call(this, pickedResumeToken)) {
+          this.set(pickedResumeToken);
+        }
       }
     }
   };
+
+  function isResumeTokenValid (resumeToken) {
+    if (this.resumeTokenSchema) {
+      var error = vat.validate(resumeToken, this.resumeTokenSchema).error;
+
+      if (error) {
+        if (this.sentryMetrics) {
+          if (error instanceof ReferenceError) {
+            error = authErrors.toMissingResumeTokenPropertyError(error.key);
+          } else {
+            error = authErrors.toInvalidResumeTokenPropertyError(error.key);
+          }
+
+          // HACK: Interpolate the invalid property name into the error
+          //       message. One day this will be handled automagically.
+          error.message = authErrors.toInterpolatedMessage(error);
```

Comment:
Hmm, I kind of disagree on this one so I'm going to state my reasons and see if I can persuade you to change your mind. 😄

1. `isResumeTokenValid` is private to this module so the risk seems low. Further, if we *did* start validating things twice for some reason, that seems like the kind of thing we'd like to know about, e.g. by noticing a doubling of the error rates in sentry. Silently pretending that double validation isn't happening in that case seems wrong to me.

2. It feels more logical to report the error as soon as we receive it rather than pass it back to a separate function. Vat's result object is as much a part of its interface as the function signature for `validate`. They should be kept close together for that reason.

3. The function names are much clearer when the concerns are separated this way. The callee is concerned with validating data and the caller is concerned populating data. If we made the change and wanted semantically-correct function names, we'd need something like `getValidationState` and `populateFromResumeTokenOrReportError`.

How did I do?